chatkit-core/src/main/kotlin/com/pusher/chatkit/state/Reducers.kt

```diff
@@ -0,0 +1,24 @@
+package com.pusher.chatkit.state
+
+import org.reduxkotlin.ReducerForActionType
+import org.reduxkotlin.reducerForActionType
+
+/**
+ * Create a reducer for the whole State
+ */
+internal inline fun <reified TAction> stateReducer(
```

Comment:
nice one, though wouldn't it be better for it to be private (and then at the bottom)? I can imagine another specific reducer creation fun will be added later: `subscriptionStateReducer` and only it alongside `chatStateReducer` should be exposed to the rest of the internal code